Add skip report method

bogdanburich opened the pull request:
Задача: https://www.notion.so/Bot-a40e24077d7f4a4aa684f979b376b090

- добавлен статус skipped в модель Report
- добавлен метод в ReportService для изменения статуса отчета
- добавлена проверка в методе send_report, что отчет не пропущен и вызов исключения, в случае, если он пропущен
- добавлена обработка исключения в photo_handler - после пропуска фотографии не принимаются
- добавлен метод для обработки кнопки "Пропустить задание"

Commit:
Add skip report method

ADesBiysk commented on src/bot/handlers.py:
```diff
@@ -149,12 +152,17 @@ async def balance(telegram_id: int) -> int:
+    """Метод для пропуска задания."""
+    session_gen = get_session()
+    session = await session_gen.asend(None)
+    user_service = UserService(UserRepository(session), RequestRepository(session))
```
Для UserService нужен ещё ShiftService.

ADesBiysk commented on src/bot/handlers.py:
```diff
@@ -149,12 +152,17 @@ async def balance(telegram_id: int) -> int:
+    session_gen = get_session()
+    session = await session_gen.asend(None)
+    user_service = UserService(UserRepository(session), RequestRepository(session))
+    report_service = ReportService(ReportRepository(session), ShiftRepository(session), MemberRepository(session))
```
Для ReportService нужен ещё TaskService.

ADesBiysk commented on src/bot/handlers.py:
```diff
@@ -149,12 +152,17 @@ async def balance(telegram_id: int) -> int:
+    user_service = UserService(UserRepository(session), RequestRepository(session))
+    report_service = ReportService(ReportRepository(session), ShiftRepository(session), MemberRepository(session))
+    user = await user_service.get_user_by_telegram_id(chat_id)
+    report = await report_service.get_current_report(user.id)
```
Метод может вызывать исключение CurrentTaskNotFoundError. Нужно обработать.

ADesBiysk commented on src/core/services/report_service.py:
```diff
@@ -83,6 +89,12 @@ async def decline_report(self, report_id: UUID, bot: Application) -> ReportRespo
 
+    async def skip_report(self, report_id: UUID) -> Report:
+        """Задание пропущено: изменение статуса."""
+        report = await self.__report_repository.get(report_id)
```
В photo_handler уже есть сущность Report. Зачем здесь её заново получать и потом уже изменять?
Здесь нужно сделать метод, который по  user_id изменяет статус отчёта. Причём, нужно проверить, что отчёт имеет статус 'waiting'. При других статусах его изменять запрещено.

ADesBiysk commented on src/core/services/report_service.py:
```diff
@@ -128,6 +140,7 @@ async def get_current_report(self, user_id: UUID) -> Report:
         return await self.__report_repository.get_current_report(user_id)
 
     async def send_report(self, report: Report, photo_url: str) -> Report:
+        await self.check_report_skipped(report.id)
```
Тоже странно. Уже есть сам Report, зачем передавать только id в check_report_service и там его заново запрашивать из БД. Нужно передавать весь Report.

ADesBiysk commented on src/core/db/migrations/versions/2023-02-23_21.17.14_add_report_status_skipped.py:
```diff
@@ -0,0 +1,27 @@
+revision = '093cd892a57e'
+down_revision = 'fb0616f8ae8e'
+branch_labels = None
+depends_on = None
```
Alembic не может автоматически корректно обработать данную ситуацию. Посмотри пример как правильно добавлять статус: https://github.com/Studio-Yandex-Practicum/lomaya_baryery_backend/blob/develop/src/core/db/migrations/versions/2023-02-07_21.11.28_add_new_status_to_shift_model.py

bogdanburich committed:
Update current branch with develop

commit 85882cf372d4640990e7e2ee219da64e27ac1a6d
Merge: 5f62044 8762464
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Wed Mar 1 02:40:17 2023 +0700

    Merge pull request #234 from Studio-Yandex-Practicum/fix/telegram_bot_error_handler

    Added errors to handle

commit 5f62044ea121c12d334627c2aeac4ba42ee03407
Merge: 09503c1 e8f1a1a
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Wed Mar 1 02:12:31 2023 +0700

    Merge pull request #235 from Studio-Yandex-Practicum/fix/poetry_version_in_workflow

    Fix version of poetry

commit e8f1a1a0607fc01d9ef517353f45b2203ab451fe
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Wed Mar 1 02:09:38 2023 +0700

    Fix version of poetry

commit 8762464af54ced5ffdeb75d3aa0d7f2958cf3b2f
Author: Alexey-Bormotov <di-devil@yandex.ru>
Date:   Tue Feb 28 17:14:43 2023 +0700

    Added errors to handle

commit 09503c1f4bbe0cec3431a57f75357c62ec9a7315
Merge: 600013e f433c45
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Tue Feb 28 04:14:59 2023 +0700

    Merge pull request #173 from Studio-Yandex-Practicum/dev_ivan_feature_forms

    Add autofill forms when user registered in project already

commit 600013ef0815e418c2c4094fab77ef17df35fe72
Merge: 94486ac 8f2f557
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Tue Feb 28 03:29:35 2023 +0700

    Merge pull request #223 from Studio-Yandex-Practicum/feature/swagger_authorization_cusomization

    Авторизация в админке изменена на вход по access токену

commit 94486ac23a07711bbbe7ce37cdaf1348e8ef22b6
Merge: 53da638 ab416b7
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Tue Feb 28 02:33:37 2023 +0700

    Merge pull request #233 from Studio-Yandex-Practicum/fix/start_shift

    fix automatically start shift

commit ab416b7edc8213a0bd01e83c62b9e365f9b0d9c4
Author: Анастасия <fedyaevaAS@yandex.ru>
Date:   Mon Feb 27 16:35:51 2023 +0300

    fix automatically start shift

commit 8f2f557610528c45ea27c05c559876fa2287ed0f
Author: Владислав Смаилов <vladislav.smailov@yandex.ru>
Date:   Sun Feb 26 15:15:28 2023 +0500

    изменена авторизация для защищенных эндпоинтов на авторизацию по токену

commit 53da63818d8679c3237dd0f7b09babac9df2d871
Merge: 480ccee d8fe4bb
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Sun Feb 26 04:52:28 2023 +0700

    Merge pull request #231 from Studio-Yandex-Practicum/fix/mail_sending

    url fix. change workflow trigger

commit d8fe4bbbc525b6e67edd2d53c1c1eeb6550d0b74
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 04:52:03 2023 +0700

    change workflow trigger

commit 64a657ecc1d3696f64a9e702bdc0e7a637eb88d8
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 04:44:42 2023 +0700

    url fix. change workflow trigger

commit 480cceed747bb60dfda0b4ad6b57dfbd7ee35619
Merge: e04e99c febefa2
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Sun Feb 26 04:24:21 2023 +0700

    Merge pull request #230 from Studio-Yandex-Practicum/fix/mail_sending

    Add test deploy

commit febefa2416c4b2bdf206c0d4051a377f096c517c
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 04:20:55 2023 +0700

    Add test deploy

commit 126110a5e92fe6e8a413f35887952a233eb738b0
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 04:15:50 2023 +0700

    Add test deploy

commit e04e99c7f3742029a349b6260f7fd85529286c0e
Merge: ae38146 1ca72be
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Sun Feb 26 03:44:05 2023 +0700

    Merge pull request #229 from Studio-Yandex-Practicum/fix/mail_sending

    Change default values

commit 1ca72bea1920169d14b518be173818945c918c97
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 03:41:39 2023 +0700

    Change default values

commit ae381469cff553a5d98199a564f437d58c54b19c
Merge: d2d857e 8eb1744
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Sun Feb 26 02:28:15 2023 +0700

    Merge pull request #228 from Studio-Yandex-Practicum/fix/mail_sending

    fix: add default settings for yandex.mail. Fix some bugs in mail send…

commit 8eb1744830776ef35cc676496c948ee60080150b
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 02:23:56 2023 +0700

    Some fix

commit a59a9e6f16ed39afd9aa9ba582860404953d800b
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 02:15:34 2023 +0700

    Add settings.py

commit dd8cb840268538b89e0ba1a973141b2e054ccf41
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 02:05:19 2023 +0700

    delete settings file

commit 703ac602410d9d177069533026febe50674b8f99
Author: ADesBiysk <adesbiysk@yandex.ru>
Date:   Sun Feb 26 02:02:01 2023 +0700

    fix: add default settings for yandex.mail. Fix some bugs in mail sending. Some refactoring.

commit d2d857ef890f5f210ae1f723313137e698600383
Merge: 8c75278 ba0105b
Author: DenisAbramenko <adesbiysk@yandex.ru>
Date:   Sat Feb 25 05:08:41 2023 +0700

    Merge pull request #215 from Studio-Yandex-Practicum/feature/api_add_invitation_get_method

    Feature/api add invitation get method

commit ba0105b478fb2b5dbd973ded883e93873e053ca8
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Thu Feb 23 21:44:33 2023 +0100

    Code-review fixes:

     - `expired-date` renamed to `expired-datetime`
     - method `list_all_invitations` in `AdministratorInvitationService`
       created

commit 4b47af70053951d269146af89e3ea321d6661d5a
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Thu Feb 23 21:26:41 2023 +0100

    Fix migrations order

commit 7a6b77258edf29d796471266dae6d664f79f0f5d
Merge: b643f23 8c75278
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Thu Feb 23 16:55:07 2023 +0100

    Merge branch 'develop' into feature/api_add_invitation_get_method

commit b643f231d9ef9b29bd833ca8883a5c627ac5ade0
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Thu Feb 23 16:12:58 2023 +0100

    Code review fixes:

     - `responses` have been added to AdministratorInvitationCBV post and get methods
     - typing fix in `AdministratorInvitationCBV.get_all_invitations`
     - `expired_date` renamed to `expired_datetime` in `AdministratorInvitation` model
     - `async def get_all` added to `AbstractRepository`
     - `get_all_invitations` removed from `AdministratorInvitationRepository`, utilized `get_all` instead
     - `get_tasks` removed from `TaskRepository`, utilized `get_all` instead

commit f433c45c3437ad47626e90d822fb1899bba1c35f
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Thu Feb 23 19:57:00 2023 +0600

    feat(core): create ApplicationError

    Created base error for app - `ApplicationError`.

commit dd8362f92c1e2102d6b5c498c00daab4bc4b83aa
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Thu Feb 23 19:40:49 2023 +0600

    feat(core): create `ApplicationError`

    Created base exception for app - `ApplicationError`.

commit d734c6c4994b70c6d3ca68ae19a59029f14916c8
Merge: 7c06c50 8c75278
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Thu Feb 23 18:05:58 2023 +0600

    Merge branch 'develop' of github.com:Studio-Yandex-Practicum/lomaya_baryery_backend into dev_ivan_feature_forms

commit 7c06c50b95140992e7bd82252e58f4d6f914c7a3
Merge: 3e41c1b 414bfcd
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Wed Feb 22 19:49:19 2023 +0600

    merging changes in develop

commit 3e41c1b543b7e081ef21b4faf8495b81c80db163
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Wed Feb 22 19:28:41 2023 +0600

    fix(api, bot, core): deactivating cache

    Deactivating cache in registration`s response.
    Refaqtoring the handler `web_app_data`, returned `exception.ValueError`.
    Little changes in settings.

commit d4e0b57ea51c88802e906860e971f7761a92a02a
Author: Владислав Смаилов <vladislav.smailov@yandex.ru>
Date:   Tue Feb 21 15:19:24 2023 +0500

    Авторизация в админке изменена на вход по токену

commit 4fd85a17fefff9ec5e37385be1ac354febc49fab
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Mon Feb 20 22:44:44 2023 +0600

    fix(api): change info for documentation endpoint

    Made change information for documentation of `webhook` endpoint.

commit fb86e823c80120bf41f8bccdea55ac2e5e7ff718
Merge: bee152b 83442c8
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Mon Feb 20 21:35:13 2023 +0600

    fix(api, bot)!: changes routers, registration form

    Made changes to registration_form & webhook routers.
    Change registration template.
    Made changes `settings` about url of webhook & path to registration template.

commit bee152b7ab07b5695854c88e886954558fff04b0
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Thu Feb 16 18:26:16 2023 +0600

    fix(bot): fixed bug in docstring

commit 5e80545a7a11d509019da354ff06517893662ea2
Merge: 219fd42 525deca
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Thu Feb 16 12:11:41 2023 +0100

    Merge branch 'develop' into feature/api_add_invitation_get_method

commit 219fd4239829f65d18294e39b5f78f060634bd62
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Thu Feb 16 12:01:43 2023 +0100

    Code review fixes:

     - removed the `role` field from API response
     - added the `expired_datetime` to API response
     - refactored: method get_invitaions renamed to get_all_invitations

commit 2fb91733814551a11af41201ad3482388db22d23
Merge: adaf680 58a82ad
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Tue Feb 14 14:35:39 2023 +0100

    Merge branch 'develop' into feature/api_add_invitation_get_method

commit adaf6807f197ef1c2da4adf79fb5844d816a1fc5
Author: Ivan Reznichenko <ivreznic@gmail.com>
Date:   Tue Feb 14 14:07:59 2023 +0100

    Get all Invitations feature has been added

     - `api/response_models/administrator_invitation.py`: file was created with AdministratorInvitationResponse model
     - `api/routers/administrator_invitation.py`: endpoint '/invite' was changed to '/invitations' in router.post
     - `api/routers/administrator_invitation.py`: method `get_all_invitations` was added
     - `core/services/administrator_invitation.py`: method `list_all_invitations` was added
     - `core/db/repository/administrator_invitation.py`: method `get_invitaions` was added

commit e80a55380ab539ad1086f46e667773d45b3ab8ea
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Mon Feb 13 21:14:43 2023 +0600

    fix(bot, templates): save user instance in context.user_data

commit 28779312e0af8c49f18c854527c62ffc567de55a
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Mon Feb 13 19:29:20 2023 +0600

    remove from PR poetry.lock

commit 148270efe46bb47f225470d8f07fb8232afbc7ec
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Mon Feb 13 18:54:48 2023 +0600

    choice autocrlf on false

commit b5202967f8a5c2a46289decf49e0d52b4448fe08
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Sun Feb 12 11:26:25 2023 +0600

    fix(api, bot, templates): fix comments PR

commit 698d4443069371e852708c742c31b17ac7a28957
Author: Ilya Steblyan <102170248+Hrushon@users.noreply.github.com>
Date:   Sun Feb 12 10:53:13 2023 +0600

    Update src/bot/handlers.py

    Co-authored-by: DenisAbramenko <adesbiysk@yandex.ru>

commit 3fd44715760608248e4995f7a6b290dc109af859
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Sat Feb 11 14:22:54 2023 +0600

    fix(bot): litle change order of messages to output

commit bb666271bbf094768d4b4e83bda176a7ee6c8467
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Sat Feb 11 13:35:04 2023 +0600

    fix(api, bot, templates): elimination of comments PR

commit 61c6e93a055a8fcc754f3a53df8ae764bf295e1a
Merge: 28ac01a a634209
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Sat Feb 11 12:00:00 2023 +0600

    Merge branch 'develop' of github.com:Studio-Yandex-Practicum/lomaya_baryery_backend into dev_ivan_feature_forms

commit 28ac01a270ec4346139fba28274208b1369c98ae
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Tue Feb 7 17:08:40 2023 +0600

    fix workflow errors

commit 25a8cb3acdbf0493bf78386458955fdf2ed62e51
Merge: 24d6a7a e3faa03
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Tue Feb 7 16:59:05 2023 +0600

    pull origin and merge conflict

commit 24d6a7a110041e674193d9534fb7dc566cb5532d
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Tue Feb 7 16:36:38 2023 +0600

    fix dependencies

commit 1ab5c9057af946b3ac082999399c3d0d781cbaff
Author: Ilia Steblyan <stebljan@yandex.ru>
Date:   Tue Feb 7 16:21:53 2023 +0600

    fixed bugs with registration form

commit 2ffbb984c592abeacff1d8021840a088d6ae8de6
Merge: 122411a f234166
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Mon Jan 30 02:14:44 2023 +0300

    Обновление регистрационных данных

commit 122411ab1e71670f3c844fa5a838b6a59c926fb0
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Sun Jan 22 21:37:34 2023 +0300

    fixes

commit 97476409fbdd6c5c95e11a1438bb369afacb2177
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Sun Jan 22 21:35:06 2023 +0300

    fixes

commit 367cf6b371bbccad649e6a12a7ca30d4983c3f43
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Sun Jan 22 21:01:14 2023 +0300

    review_fixes

commit 44ca4d807076afb6d13a4c1f650fab3ec9ca860c
Merge: 05f0948 9842fee
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Sun Jan 22 20:53:42 2023 +0300

    review_fixes

commit 05f094826545ffe10ed814b35f203f1843f306ea
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Sun Jan 22 20:48:57 2023 +0300

    review_fixes

commit e232da32f35222258cb9c0a773eb8c2b19794860
Merge: 8c522b9 1d4138f
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Tue Jan 17 02:39:00 2023 +0300

    fixes

commit 8c522b9cd42507edf4f34e16a091c7df0caa7707
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Tue Jan 17 02:30:58 2023 +0300

    fixes

commit 6ac3cefad8c12759375382735a4d2af78e4ef331
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Tue Jan 17 02:30:21 2023 +0300

    fixes

commit 8294018e2b4d5de35fdd04e1deebbe858b3da6b1
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Fri Jan 13 13:35:25 2023 +0300

    add jinja2

commit 6170cb250ac17bbcf5926d525d85d5cc75d11bd8
Merge: ec839e4 6b3757f
Author: Ninefiveblade <cherep1.92@yandex.ru>
Date:   Fri Jan 13 13:04:10 2023 +0300

    Merge branch 'develop' of github.com:Studio-Yandex-Practicum/lomaya_baryery_backend into dev_ivan_feature_forms

commit ec839e4ee422f313d01f41dcfc77935fb6a28d63
Merge: a010dd7 fe7150c
Author: Anton Zelinsky <Anton.Zelinsky@gmail.com>
Date:   Tue Jan 3 23:31:49 2023 +0100

    Merge pull request #167 from Studio-Yandex-Practicum/fix/report_attempt_number

    changed report field is_repeated to attempt_number

bogdanburich committed:
Update migration flow

bogdanburich committed:
Fixes after review

bogdanburich committed:
Merge branch 'develop' into feature/skip_report_button

bogdanburich committed:
Delete extra change

bogdanburich commented:
Не уверен, что правильным способом получил изменения из develop, но вроде как все в порядке, замечания исправил

bogdanburich committed:
Fix error

ADesBiysk commented on src/core/services/report_service.py:
```diff
@@ -83,6 +88,14 @@ async def decline_report(self, report_id: UUID, bot: Application) -> ReportRespo
         await self.__notify_member_about_finished_shift(member, bot)
         return report
 
+    async def skip_report(self, user_id: UUID) -> Report:
```
Имя не информативное имя метода. Не понятно, какой отчёт пропустить. Лучше: skip_current_report.

ADesBiysk commented on src/bot/handlers.py:
```diff
@@ -163,14 +166,19 @@ async def photo_handler(update: Update, context: CallbackContext) -> None:
-        await skip_task(update.effective_chat.id)
+        try:
+            await skip_report(update.effective_chat.id)
+        except CurrentTaskNotFoundError:
```
Исключение ReportAlreadyReviewedException не обработано.

bogdanburich committed:
Fixes after review

Ivan-Skvortsov commented on src/core/services/report_service.py:
```diff
@@ -83,6 +88,14 @@ async def decline_report(self, report_id: UUID, bot: Application) -> ReportRespo
+    async def skip_current_report(self, user_id: UUID) -> Report:
+        """Задание пропущено: изменение статуса."""
+        report = await self.__report_repository.get_current_report(user_id)
+        if report.status is not Report.Status.WAITING:
```
Если мы скипнем репорт и затем снова нажмём пропустить - это условие тоже выполнится, т.к. у репорта будет статус SKIPPED. 
Надо бы это тоже учесть - иначе в таком случае пользователь получит неправильное сообщение (что-то вроде "ранее отправленный отчет проверяется...").

Ivan-Skvortsov commented on src/bot/handlers.py:
```diff
@@ -163,14 +167,23 @@ async def photo_handler(update: Update, context: CallbackContext) -> None:
+            await update.message.reply_text("Сейчас заданий нет.")
+        except ReportAlreadyReviewedException:
+            await update.message.reply_text(
+                "Ранее отправленный отчет проверяется или уже принят, нельзя пропустить отчет."
```
```suggestion
                "Ранее отправленный отчет проверяется или уже принят, сейчас нельзя пропустить задание."

```

Ivan-Skvortsov commented on src/bot/handlers.py:
```diff
@@ -163,14 +167,23 @@ async def photo_handler(update: Update, context: CallbackContext) -> None:
     if update.message.text == LOMBARIERS_BALANCE:
         amount = await balance(update.effective_chat.id)
         await update.message.reply_text(f"Количество ломбарьеров = {amount}.")
     elif update.message.text == SKIP_A_TASK:
```
А почему `elif`? Судя по коду, тут достаточно `if`

bogdanburich replied:
Эту часть внимательно не смотрел, честно говоря, тк этот код уже был
Но тут действительно можно обойтись if вместо elif, поправлю

Ivan-Skvortsov commented on src/bot/handlers.py:
```diff
@@ -182,8 +195,18 @@ async def balance(telegram_id: int) -> int:
+        ReportRepository(session), ShiftRepository(session), MemberRepository(session), task_service
+    )
+    user = await user_service.get_user_by_telegram_id(chat_id)
+    await report_service.skip_current_report(user.id)
```
Было бы хорошо после пропуска дать фидбек пользователю, что задание пропущено и следующее придёт в 8 утра)

bogdanburich committed:
Update text

Co-authored-by: Ivan Skvortsov <pprofcheg@gmail.com>

bogdanburich committed:
Fixes after review

bogdanburich committed:
Merge remote-tracking branch 'origin/develop' into feature/skip_report_button

bogdanburich committed:
Recreate migration
